### ISSUE TITLE
Add Gaussian Process Approximation (GPA) layer

### ISSUE THREAD

**Issue by hasan2m** [JeffersonLab/jlab_datascience_core#71]
We are using GPA for quantifying uncertainty in many of our projects. It is better to have a version of GP layer in the core so far anyone can import that from the core if needed.

- [x] Create an issue and branch
- [x] Add GP layer and Negative log likelihood loss function
- [x] Add unit test
- [x] Create a Pull request

**Comment by sgoldenCS** [comment]
Maybe there's a way to integrate the code from this repo? There's also a notebook in that repo with an example of using it.
https://github.com/JeffersonLab/jlab_datascience_dev/blob/main/jlab_datascience_dev/keras_models/gaussian_process_layer.py

**Comment by hasan2m** [comment]
Added GP layer and negative log-likelihood loss function in the utils.

Unit test passed for GP Layer: 

![Image](https://github.com/user-attachments/assets/7dbd02a4-173f-42b3-8c59-1f92064e7012)

Unit test passed for loss function: 

![Image](https://github.com/user-attachments/assets/e3eeb3ba-f6ca-4c98-8052-2a3d079b7cf9)
